pkg/build/admission/overrides/admission.go

```diff
@@ -0,0 +1,98 @@
+package overrides
+
+import (
+	"io"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	buildadmission "github.com/openshift/origin/pkg/build/admission"
+)
+
+func init() {
+	admission.RegisterPlugin("BuildOverrides", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		overridesConfig, err := getConfig(config)
+		if err != nil {
+			return nil, err
+		}
+
+		glog.V(4).Infof("Initializing BuildOverrides plugin with config: %#v", overridesConfig)
+		return NewBuildOverrides(overridesConfig), nil
+	})
+}
+
+func getConfig(in io.Reader) (*BuildOverridesConfig, error) {
+	overridesConfig := &BuildOverridesConfig{}
+	err := buildadmission.ReadPluginConfig(in, overridesConfig)
+	if err != nil {
+		return nil, err
+	}
+	return overridesConfig, nil
+}
+
+type buildOverrides struct {
+	*admission.Handler
+	overridesConfig *BuildOverridesConfig
+}
+
+// NewBuildOverrides returns an admission control for builds that overrides
+// settings on builds
+func NewBuildOverrides(overridesConfig *BuildOverridesConfig) admission.Interface {
+	return &buildOverrides{
+		Handler:         admission.NewHandler(admission.Create, admission.Update),
+		overridesConfig: overridesConfig,
+	}
+}
+
+// Admit appplies configured overrides to a build in a build pod
+func (a *buildOverrides) Admit(attributes admission.Attributes) error {
+	if a.overridesConfig == nil {
+		return nil
+	}
+	if !buildadmission.IsBuildPod(attributes) {
+		return nil
+	}
+	return a.applyOverrides(attributes)
+}
+
+func (a *buildOverrides) applyOverrides(attributes admission.Attributes) error {
+	if !a.overridesConfig.ForcePull {
```

Comment:
i imagine we will add stuff at some point, but i don't think it's going to be any more work to refactor it then, than it would be to refactor it now.